Carbon.Sample.API/Program.cs

```diff
@@ -1,23 +1,70 @@
+using Carbon.Redis;
+using Carbon.Sample.API.Infrastructure.Contexts.SampleContext;
 using Carbon.WebApplication;
+using Carbon.WebApplication.EntityFrameworkCore;
+using Carbon.WebApplication.SolutionService;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using Carbon.Sample.API.Infrastructure;
+using Carbon.Sample.API.Infrastructure.Extensions;
+using Carbon.HttpClient.Auth;
+using Carbon.WebApplication.Middlewares;
+using Carbon.WebApplication.SolutionService.Domain;
+using Carbon.WebApplication.TenantManagementHandler.Dtos.ErrorHandling;
+using Microsoft.Extensions.Configuration;
 
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Hosting;
+Console.Title = @"CARBON_SAMPLE_API";
+const string cors = "_carbon-sample";
+var builder = WebApplication.CreateBuilder(args);
 
-namespace Carbon.Sample.API
+builder.AddCarbonServices((services) =>
 {
-	public class Program
-	{
-		public static void Main(string[] args)
-		{
-			CreateHostBuilder(args).Build().Run();
-		}
-
-		public static IHostBuilder CreateHostBuilder(string[] args) =>
-			Host.CreateDefaultBuilder(args)
-				.ConfigureWebHostDefaults(webBuilder =>
-				{
-					webBuilder.UseCarbonFeatures<Startup>();
-					webBuilder.UseStartup<Startup>();
-				});
-	}
-}
+    services.AddBearerAuthentication(builder.Configuration);
+    services.AddLocalization(o => { o.ResourcesPath = "Resources"; });
+    // :TODO carbon
+    services.AddCustomCors(cors);
+    services.AddRedisPersister(builder.Configuration);
+    services.AddDatabaseContext<SampleDBContext, Program>(builder.Configuration);
+    services.AddApplicationDependencies();
+    services.AddHttpClientAuth();
+    //Enables Solution based saga events for feature list etc
+    services.ConfigureAsSolutionService(builder.Configuration);
+    return services;
+});
+var app = builder.Build();
+app.AddCarbonApplication((application) =>
+{
+    application.UseCors(cors);
+    application.UseHsts();
+    application.UseHttpsRedirection();
+    application.UseCustomRequestLocalization();
+    app.CreateAuthentication("RAM");
```

Comment:
Burada RAM kısmı için bir açıklama ekleyelim; neden buraya RAM yazdı vb bilmesi için